mkosi/kmod.py

```diff
@@ -18,41 +20,101 @@
 def loaded_modules() -> list[str]:
     # Loaded modules are listed with underscores but the filenames might use dashes instead.
     return [
-        rf"/{line.split()[0].replace('_', '[_-]')}\.ko"
-        for line in Path("/proc/modules").read_text().splitlines()
+        normalize_module_name(line.split()[0]) for line in Path("/proc/modules").read_text().splitlines()
     ]
 
 
+def globs_match_filename(name, globs) -> Optional[bool]:
+    # Check whether the path matches any of the globs
+
+    for glob in reversed(globs):
+        # Patterns are evaluated in order and last matching one wins.
+        # Patterns may be prefixed with '-' to exclude modules.
+        if negative := glob.startswith("-"):
+            glob = glob[1:]
+        # As a special case, if a directory is specified, all items
+        # below that directory are matched.
+        if glob.endswith("/"):
+            glob += "*"
+
+        if (
+            # match the full path
+            (glob.startswith("/") and fnmatch.fnmatch(f"/{name}", glob))
+            or
+            # match a subset of the path, at path element boundary
+            ("/" in glob and fnmatch.fnmatch(f"/{name}", f"*/{glob}"))
+            or
+            # match the basename
+            fnmatch.fnmatch(name.split("/")[-1], glob)
+        ):
```

Comment:
This can be compressed by not having the comments between the or and the following predicate, that these are ored is already sufficiently highlighted from it leading the line.
```suggestion
        if (
            # match the full path
            (glob.startswith("/") and fnmatch.fnmatch(f"/{name}", glob))
            # match a subset of the path, at path element boundary            
            or ("/" in glob and fnmatch.fnmatch(f"/{name}", f"*/{glob}"))
            # match the basename            
            or fnmatch.fnmatch(name.split("/")[-1], glob)
        ):
```